src/js/references/hooks.ts

```diff
@@ -128,3 +131,44 @@ export function useUpdateSourceTypes(
         register,
     };
 }
+
+/**
+ * All reference rights
+ */
+export enum ReferenceRight {
+    modify = "modify",
+    modify_otu = "modify_otu",
+}
+
+/**
+ * Check if the logged in account has the passed `right` on the reference detail is loaded for.
+ *
+ * @param referenceId - The id of the reference to check
+ * @param right - The right to check for (eg. modify_otu)
+ * @returns Whether the right is possessed by the account
+ */
+export function useCheckReferenceRight(referenceId: string, right: ReferenceRight) {
+    const queryClient = useQueryClient();
+    const { data: account, isLoading: isLoadingAccount } = useFetchAccount();
+
+    const reference: Reference = queryClient.getQueryData([...referenceQueryKeys.all(), referenceId]);
+
+    if (isLoadingAccount || !reference) {
```

Comment:
```suggestion
    if (isLoadingAccount || isLoadingReference) {
```
With the tweak above